### PR TITLE
Switch @testing-library/dom to be a peerDep + devDep

### LINKED ISSUE #135: Allow for non-fixed version of @testing-library/dom
Hi, thanks for this package, it's pretty awesome :eyes: :tada: 

I was wondering if the `package.json` could be updated to have a semver _range_ for the `@testing-library/dom` dependency, rather than a fixed version: https://github.com/testing-library/user-event/blob/master/package.json#L56

Suggestion:

```diff
-    "@testing-library/dom": "5.1.1"
+    "@testing-library/dom": "^5.1.1"
```

This way I don't have to have multiple versions of `@testing-library/dom` installed, and you don't have to constantly update the version in your `package.json` :smile: 

Happy to submit a PR if you want

### PR DESCRIPTION
**Pull request by calebeby**
Closes #135

### CODE REVIEW

**Comment by NMinhNguyen** [comment]
Strictly speaking, this could be considered a breaking change - because what was being installed before, now has to be installed separately. You might need to update the docs to reflect the peer dependency requirement as well, similar to how React Testing Library mentions `react` and `react-dom` as `peerDependencies`: https://github.com/testing-library/react-testing-library#installation

**Comment by calebeby** [comment]
Maybe we could keep it as a dependency, but switch it to a version range, like I originally suggested, so it's not a breaking change 😄

**Comment by NMinhNguyen** [comment]
@kentcdodds @Gpx do you guys have a preference on which approach to take here? :slightly_smiling_face:

File: package.json [hunk 1]
```diff
@@ -52,8 +53,8 @@
     "semantic-release": "15.13.12",
     "travis-deploy-once": "5.0.11"
   },
-  "dependencies": {
-    "@testing-library/dom": "5.1.1"
+  "peerDependencies": {
+    "@testing-library/dom": "^5.1.1"
```
Comment (kentcdodds) [inline package.json]:
```suggestion
    "@testing-library/dom": ">=5"
```

Comment (kentcdodds) [reply package.json]: Peer Dependencies should have a version range that's as permissible as possible. That way people can install their own version without conflicting with the one this package says it needs.

Comment (calebeby) [reply package.json]: @kentcdodds what if `@testing-library/dom` releases a v6 that has breaking changes that affect this package? That's why I went with a caret range. Maybe the caret range should be `^5.0.0` though?

Comment (kentcdodds) [reply package.json]: Chances are that a breaking change in `@testing-library/dom` wont impact this package. If it does, then we can fix this. The last thing I want is to add to the already ridiculous amount of warnings people get when they install deps in their project.

**Comment by Gpx** [comment]
Thanks to everyone 🎉

**Pull request merged**